salt/certbot.sls

```diff
@@ -0,0 +1,36 @@
+# certbot.sls
+#  installs certbot-auto
+#  see also 'certbot' macro in lib.sls
+#
+#  NOTES
+#
+#  * We would prefer to use a proper distro repo instead of this :(
+#  * We grab a specific commit 'ce4e00569e6d8ed3d51c5a078d4281bec5f8e5f0'
+#    from the upstream git repo -- salt wants a stable source-hash
+#    to be cool about downloading from the net, and frankly I agree.
+#    (See https://github.com/certbot/certbot/releases where 'ce4e005'
+#    is the commit hash of tag v0.9.3)
+#  * certbot-auto downloads and installs a shedload of backport stuff via
+#    apt-get: augeas python python-dev virtualenv gcc dialog libssl-dev
+#             libffi-dev ca-certificates
+#    and also python deps into /root/.local/share/letsencrypt/
+
+certbot-auto:
+  file.managed:
+    - name: /root/certbot-auto
+    - source: https://raw.githubusercontent.com/certbot/certbot/ce4e00569e6d8ed3d51c5a078d4281bec5f8e5f0/certbot-auto
+    - source_hash: sha256=6249576909473ffddd945f8574b4035fd4a2be0323f748a650565ae32b9d4971
+    - mode: 755
+
+certbot-initialise:
+  cmd.run:
+    - name: /root/certbot-auto plugins --non-interactive
+    - require:
+      - file: /root/certbot-auto
+
+certbot-cron-renewal:
+  cron.present:
```

Comment:
I would add an identifier to this so we don't get duplicates if the command changes slightly - https://docs.saltstack.com/en/latest/ref/states/all/salt.states.cron.html ctrl-f identifier